sdks/python/apache_beam/yaml/yaml_enrichment.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Dict
+import apache_beam as beam
+from apache_beam.transforms.enrichment_handlers.bigquery import BigQueryEnrichmentHandler
+from apache_beam.transforms.enrichment_handlers.bigtable import BigTableEnrichmentHandler
+from apache_beam.transforms.enrichment_handlers.feast_feature_store import FeastFeatureStoreEnrichmentHandler
+from apache_beam.transforms.enrichment_handlers.vertex_ai_feature_store import VertexAIFeatureStoreEnrichmentHandler
+from apache_beam.transforms.enrichment import Enrichment
+from typing import Optional
+
+@beam.ptransform.ptransform_fn
+def enrichment_transform(pcoll, enrichment_handler: str, handler_config: Dict[str, Any], timeout: Optional[float] = 30):
+    """
+    The Enrichment transform allows you to dynamically enhance elements in a pipeline 
+    by performing key-value lookups against external services like APIs or databases. 
+
+    Args:
+        enrichment_handler: Specifies the source from where data needs to be extracted
+            into the pipeline for enriching data. It can be a string value in ["BigQuery", 
+            "BigTable", "FeastFeatureStore", "VertexAIFeatureStore"].
+        handler_config: Specifies the parameters for the respective enrichment_handler in a dictionary format. 
+            BigQuery:  project, table_name, row_restriction_template, fields, column_names, condition_value_fn, query_fn, min_batch_size, max_batch_size
+            BigTable: project_id, instance_id, table_id row_key, row_filter, app_profile_id, encoding, ow_key_fn, exception_level, include_timestamp
+            FeastFeatureStore: feature_store_yaml_path, feature_names, feature_service_name, full_feature_names, entity_row_fn, exception_level
+            VertexAIFeatureStore: project, location, api_endpoint, feature_store_name:, feature_view_name, row_key, exception_level
```

Comment:
Could be convenient to link to PyDocs for the different handlers:

For example:
https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.enrichment_handlers.bigquery.html
https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.enrichment_handlers.bigtable.html
https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.enrichment_handlers.feast_feature_store.html
https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.enrichment_handlers.vertex_ai_feature_store.html